addOns/automation/src/main/java/org/zaproxy/addon/automation/tests/UrlPresenceTest.java

```diff
@@ -0,0 +1,250 @@
+                                e.getMessage()));
+            }
+        }
+        return null;
+    }
+
+    public boolean matchPattern(
+            String regex, HttpMessage msg, String param, AutomationProgress progress) {
+        Pattern pattern = compilePattern(regex, progress);
+        boolean match = false;
+        if (pattern != null) {
+            switch (param) {
+                case "requestHeaderRegex":
+                    String requestHeader = msg.getRequestHeader().toString();
+                    match = pattern.matcher(requestHeader).find();
+                    break;
+                case "requestBodyRegex":
+                    String requestBody = msg.getRequestBody().toString();
+                    match = pattern.matcher(requestBody).find();
+                    break;
+                case "responseHeaderRegex":
+                    String responseHeader = msg.getResponseHeader().toString();
+                    match = pattern.matcher(responseHeader).find();
+                    break;
+                case "responseBodyRegex":
+                    String responseBody = msg.getResponseBody().toString();
+                    match = pattern.matcher(responseBody).find();
+                    break;
+                default:
+                    throw new RuntimeException("Unknown param: " + param);
+            }
+        }
+        return match;
+    }
+
+    @Override
+    public boolean runTest(AutomationProgress progress) {
+        try {
+            URI uri = new URI(this.getData().getUrl(), true);
+            SiteMap tree = Model.getSingleton().getSession().getSiteTree();
+            SiteNode node = tree.findNode(uri);
+
+            HttpMessage msg = node.getHistoryReference().getHttpMessage();
+            if (msg != null) {
+                String requestHeaderRegex = this.getData().getRequestHeaderRegex();
+                String requestBodyRegex = this.getData().getRequestBodyRegex();
+                String responseHeaderRegex = this.getData().getResponseHeaderRegex();
+                String responseBodyRegex = this.getData().getResponseBodyRegex();
+
+                if (isNullOrEmpty(requestHeaderRegex)
+                        && isNullOrEmpty(requestBodyRegex)
+                        && isNullOrEmpty(responseHeaderRegex)
+                        && isNullOrEmpty(responseBodyRegex)) {
+                    progress.error(
+                            Constant.messages.getString(
+                                    "automation.tests.url.noregex",
+                                    getJobType(),
+                                    getName(),
+                                    getJobType()));
+                    return false;
+                }
+
+                if (!isNullOrEmpty(requestHeaderRegex)) {
+                    if (!matchPattern(
+                            requestHeaderRegex, msg, PARAM_REQUEST_HEADER_REGEX, progress)) {
+                        return false;
```

Comment:
Should these really be returning false as soon as one doesn't match?

Reply:
I guess so ... Like If two regexes are provided ... say One for body and one for head.. the body matches but head doesn't .. so it should return false ... right?

Reply:
I'm not sure. I kind of thought it was an any match situation. Maybe @psiinon can clarify his vision.

Reply:
We've just discussed this in IRC :)
We've agreed there should be a new field "operator" which can be "AND" (the default) or "OR" - that way it can be explicitly set by the user.

Reply:
got it.